src/Meilisearch/Index.Documents.cs

```diff
@@ -355,7 +355,7 @@ public async Task<T> GetDocumentAsync<T>(int documentId, CancellationToken cance
         /// <param name="cancellationToken">The cancellation token for this call.</param>
         /// <typeparam name="T">Type of the document.</typeparam>
         /// <returns>Returns the list of documents.</returns>
-        public async Task<IEnumerable<T>> GetDocumentsAsync<T>(DocumentQuery query = default,
+        public async Task<ResourceResults<IEnumerable<T>>> GetDocumentsAsync<T>(DocumentQuery? query = default,
```

Comment:
In line 346 https://github.com/meilisearch/meilisearch-dotnet/pull/290/files#diff-b99cc3e20529ed19522ca88352dc425119bb704d9bf108d5dace1ef05b8e1941L346 you can add the `List<string> fields?` as an argument as well.